src/v/raft/consensus.cc

```diff
@@ -3897,8 +3917,30 @@ std::optional<model::offset> consensus::get_learner_start_offset() const {
 
 void consensus::notify_config_update() {
     _write_caching_enabled = log_config().write_caching();
-    _flush_bytes = log_config().flush_bytes();
-    _flush_ms = log_config().flush_ms();
+    _max_pending_flush_bytes = log_config().flush_bytes();
+    _max_flush_delay_ms
+      = flush_jitter_t{log_config().flush_ms(), flush_ms_jitter}
+          .next_duration();
+    // let the flusher know that the tunables have changed, this may result
+    // in an extra flush but that should be ok since this this is a rare
+    // operation.
+    _background_flusher.signal();
+}
+
+ss::future<> consensus::background_flusher() {
+    while (!_bg.is_closed()) {
+        try {
+            co_await _background_flusher.wait(log_config().flush_ms());
```

Comment:
this is a bit of a busy loop especially with high number of partitions; do we need it?

we already signal the cvar when we append, why we need the timeout?

Reply:
This implements the core of flush.ms functionality. A timer per raft group for two reasons

* flush.ms is a topic level property, so could be different for each group.
* We need to flush bytes that haven't hit flush.bytes limit but hit the flush.ms delay.

Its 100ms by default timer and only does work only if there is something to be flushed, so I doubt if it will have any substantial impact on the CPU usage but definitely something to monitor with large tiers.

Perhaps one follow up improvement could be to have one timer per topic / core, if this turns out to be a problem.